scout/parse/config_base.py

```diff
@@ -0,0 +1,262 @@
+    peddy_ped: Optional[str] = None
+    peddy_sex: Optional[str] = None
+    phenotype_terms: Optional[List[str]] = None
+    rank_model_version: Optional[str] = ""
+    rank_score_threshold: int = 0
+    samples: List[ScoutIndividual] = []
+    smn_tsv: Optional[str] = None
+    sv_rank_model_version: Optional[str] = ""
+    synopsis: Union[List[str], str] = None
+    track: Literal["rare", "cancer"] = "rare"
+    vcf_files: Optional[VcfFiles] = None
+
+    # override init() for handling nested vcf_files dicts
+    # use try/except to handle TypeError if `vcf_files`is already set in
+    # previous call `parse_case_data()`, `parse_case()`.
+    def __init__(self, **data):
+        vcfs = VcfFiles(**data)
+        try:
+            super().__init__(vcf_files=vcfs, **data)
+        except TypeError as err:
+            super().__init__(**data)
+
+    @validator("analysis_date")
+    def check_analysis_date(cls, dt):
+        if isinstance(dt, datetime.datetime):
+            # return datetime.datetime.strptime(dt, "%Y-%M-%d %H:%M%S")
+            return dt
+        correct_date = datetime.datetime.now()
+        return correct_date
+
+    # Each case has to have a owner. If not provided in config file it needs to be given as a
+    # argument
+    @validator("owner", pre=True, always=True)
+    def mandatory_check_owner(cls, value):
+        if value is None:
+            raise ConfigError("A case has to have a owner")
+        return value
+
+    @validator("family", pre=True, always=True)
+    def mandatory_check_family(cls, value):
+        if value is None:
+            raise ConfigError("A case has to have a 'family'")
+        return value
+
+    @validator("madeline_info")
+    def check_if_madeline_exists(cls, path):
+        """Add the pedigree figure, this is a xml file which is dumped in the db"""
+        mad_path = Path(path)
+        if not mad_path.exists():
+            raise ValueError("madeline path not found: {}".format(mad_path))
+        with mad_path.open("r") as in_handle:
+            return in_handle.read()
+
+    @validator("individuals")
+    def family_relations_consistent(cls, individuals):
+        individual_dicts = [i.dict() for i in individuals]
+        if len(individual_dicts) == 0:
+            raise PedigreeError("No samples could be found")
+        all_ids = [i["individual_id"] for i in individual_dicts]
+        # Check if relations are correct
+        for parsed_ind in individual_dicts:
+            father = parsed_ind.get("father")
+            if father and father != "0":
+                if father not in all_ids:
+                    raise PedigreeError("father %s does not exist in family" % father)
+            mother = parsed_ind.get("mother")
+            if mother and mother != "0":
+                if mother not in all_ids:
+                    raise PedigreeError("mother %s does not exist in family" % mother)
+        return individuals
+
+    @validator("track")
+    def field_not_none(cls, v):
```

Comment:
I don't understand why has validator("track") and then it says "Owner and family can not be None"